compiler/rustc_attr_parsing/src/attributes/cfg.rs

```diff
@@ -228,24 +231,28 @@ pub fn eval_config_entry(
             }
         }
         CfgEntry::NameValue { name, name_span, value, span } => {
-            match sess.psess.check_config.expecteds.get(name) {
-                Some(ExpectedValues::Some(values)) if !values.contains(&value.map(|(v, _)| v)) => {
-                    id.emit_span_lint(
-                        sess,
-                        UNEXPECTED_CFGS,
-                        *span,
-                        BuiltinLintDiag::UnexpectedCfgValue((*name, *name_span), *value),
-                    );
+            if emit_lints {
```

Comment:
Why is this now necessary? in particular since it wasn't necessary before.

Reply:
We didn't have one before the refactor to new attribute system, hence why I'm wondering why it's suddenly necessary?

`#[cfg]` can be quite finicky so I want to make sure we won't be missing some lints.

Reply:
Seems like we also emitted the lint twice before, weird https://rust.godbolt.org/z/8eE49bxd9.

Reply:
Yeah thanks for being thorough here, I appreciate it! Would hate to get this wrong